airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -549,6 +566,30 @@ private static void validateSchema(final RecordSchemaValidator recordSchemaValid
     }
   }
 
+  private static void populatedStreamToSelectedFields(ConfiguredAirbyteCatalog catalog,
```

Comment:
probably worth some javadoc with a brief explanation of why this achieves column selection, ie "The configured catalog only contains fields that were selected, so by iterating through each field for each stream, we identify just the fields that were selected"